apps/site/assets/ts/hooks/useUrlSearchParams.ts

```diff
@@ -0,0 +1,18 @@
+import { useEffect, useState } from "react";
+
+const useUrlSearchParams = (): URLSearchParams | null => {
+  const [searchParams, setSearchParams] = useState<URLSearchParams | null>(
+    null
+  );
+
+  const searchString = window.location.search;
+
+  useEffect(() => {
+    const urlParams = new URLSearchParams(window.location.search);
+    setSearchParams(urlParams);
+  }, [searchString]);
```

Comment:
Yeah this was my attempt at listening to the URL for updates to change which directions hours were shown.  It never worked as intended (because like you said the variable isn't tracked).  In an ideal world would could use a library such as `react-router` to monitor the URL but since this isn't a single page application that won't work.

I kept this in because it worked for initial page load.  Though you are right this could be precieved as a valid hook and be confusing to future devs if they try to use it.

I will remove it and just check the location search in the RapidTrasntit component